### DISCUSSION TITLE
Will the http headers in a source be passed through the transforms/sinks, or discarded?

### DISCUSSION THREAD

**Discussion by marcolan018** [vectordotdev/vector#10259] [Q&A] [answered]
We have a scenario to use prometheus_remote_write source, then through some transforms, finally go to a prometheus_remote_write sink.
When post data to Vector source, we will set a tenant id inside a http header and hope that header can be pass through to the sink, so that when Vector send request to the sink endpoint, the header is still there.
Is this supported?

**Comment by jszwedko** [comment, marked as answer]
Hi @marcolan018 !

Not currently, but this is something we could add if you wanted to open a feature request. We have added this sort of support to other components (like the `http` sink).
